docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,403 @@
+  - [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [When to Use `simulateTransaction` (whenever you use soroban :D )](#when-to-use-simulatetransaction-whenever-you-use-soroban-d-)
+- [How to Call `simulateTransaction`](#how-to-call-simulatetransaction)
+  - [Using Fetch](#using-fetch)
+  - [Using the JavaScript SDK](#using-the-javascript-sdk)
+- [Understanding the Footprint](#understanding-the-footprint)
+- [Assembling a Transaction](#assembling-a-transaction)
+  - [javascript-sdk-assembleTransaction](#javascript-sdk-assembletransaction)
+- [Handling Archived Ledger Entries](#handling-archived-ledger-entries)
+- [Fees and Resource Usage](#fees-and-resource-usage)
+- [Preflight Error Handling](#preflight-error-handling)
+  - [Common Preflight Errors](#common-preflight-errors)
+- [Backend Code and Workflow](#backend-code-and-workflow)
+- [Further Reading](#further-reading)
+
+## Overview
+
+The `simulateTransaction` endpoint in Soroban RPC allows you to submit a trial contract invocation to simulate how it would be executed by the Stellar network. This simulation calculates the effective transaction data, required authorizations, and minimal resource fee. It provides a way to test and analyze the potential outcomes of a transaction without actually submitting it to the network. It can be a nice way to get contract data as well sometimes.
+
+While calling the method on the rpc server is not the ONLY way to simulate a transaction, it will likely be the most common and easiest way.
+
+Here we will look at the objects involved and their definitions.
+
+## RPC Servers
+
+Stellar Development Foundation provides a testnet RPC endpoint at For Public network providers please refer to the [Ecosystem RPC Providers] list. At the time of writing this documentation there are two public and free RPC providers, and many more you can subscribe to.
+
+### Public Network Endpoints:
+
+[Liquify](https://www.liquify.com) https://stellar-mainnet.liquify.com/api=41EEWAH79Y5OCGI7/mainnet [Gateway.fm](https://www.gateway.fm) https://soroban-rpc.mainnet.stellar.gateway.fm
+
+### Testnet Endpoint:
+
+https://soroban-testnet.stellar.org:443
+
+**SimulateTransactionParams** is the argument passed to the rpc endpoint for simulateTransaction
+
+```typescript
+interface SimulateTransactionParams {
+  transaction: string; // The Stellar transaction to be simulated, serialized as a base64 string.
+  resourceConfig?: {
+    instructionLeeway: number; // Allow this many extra instructions when budgeting resources.
+  };
+}
+```
+
+**SimulateTransactionResult** is the return result from the call. It includes many useful things! [take a look at the code here!](https://github.com/stellar/stellar-docs/blob/88eeb4a318778513880e01fe9728c4cbbc73b87c/openrpc/src/methods/simulateTransaction.json#L18).
+
+## When to Use `simulateTransaction` (whenever you use soroban :D)
```

Comment:
When wouldn't I need it? If i'm fetching a ledger key directly, or doing something that doesn't require a transaction or simulation thereof?

Reply:
> Since you noted it, it might be useful to highlight a brief note about when you _wouldn't_ need simulateTransaction 😉

I changed the text to be "things simulatetransaction is used for in soroban" because there's lots of times to not use simulatetransaction, but I can't think of anything that would be relevant to bring up for this guide.  I also added a note to the section about simulating results for getter calls, that says `it's worth noting that you could sometimes get the ledger entry directly without simulation`